adam-python/src/bdgenomics/adam/adamContext.py

```diff
@@ -0,0 +1,131 @@
+#
+# Licensed to Big Data Genomics (BDG) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The BDG licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from bdgenomics.adam.rdd import AlignmentRecordRDD, \
+    FeatureRDD, \
+    FragmentRDD, \
+    GenotypeRDD, \
+    NucleotideContigFragmentRDD, \
+    VariantRDD
+    
+
+class ADAMContext(object):
+    """
+    The ADAMContext provides functions on top of a SparkContext for loading
+    genomic data.
+    """
+
+
+    def __init__(self, sc):
+        """
+        Initializes an ADAMContext using a SparkContext.
+
+        :param pyspark.context.SparkContext sc: The currently active
+        SparkContext.
+        """
+
+        self._sc = sc
+        self._jvm = sc._jvm
+        c = self._jvm.org.bdgenomics.adam.rdd.ADAMContext(sc._jsc.sc())
+        self.__jac = self._jvm.org.bdgenomics.adam.api.java.JavaADAMContext(c)
+
+
+    def loadAlignments(self, filePath):
```

Comment:
The ones with the two file paths are unnecessary after https://github.com/bigdatagenomics/adam/commit/37b971a3afb80598ffecd5275aff6c3ceaa96ef3 added `union` to GenomicRDD. That said, I think I'm missing the `union` operator in Python-land, so I'll add that.

It's hard to trace the history on `optRecordGroup`, but that codepath only exists for FASTQ. I'm of the opinion that we shouldn't add it in downstream APIs unless someone asks for it.